syntax = "proto3";

option java_multiple_files = true;
option java_package = "io.grpc.examples.cafe";
option java_outer_classname = "CafeProto";
option objc_class_prefix = "HLW";

package cafe;

// Our Cafe's interface
service Menu {
  // Returns the price of an item
  rpc Pricing (ClientRequest) returns (ProductInfo) {}

  // Returns the price of an item
  rpc SpecialOfTheDay (Weekday) returns (SpecialDrink) {}
}

message ClientRequest {
  string product = 1;
}

message Weekday {
  int32 weekday = 1;
}

message ProductInfo {
  string product = 1;
  float price = 2;
  string flavors = 3;
  enum Availability {
    Always = 0;
    Weekends = 1;
  }
  Availability availability = 4;
}

message SpecialDrink{
  string product = 1;
  float special_price = 2;
}